src/daft-core/src/array/ops/list.rs

```diff
@@ -625,6 +628,49 @@ impl ListArray {
             self.validity().cloned(),
         ))
     }
+
+    pub fn list_contains(&self, contains: &Series) -> DaftResult<BooleanArray> {
+        let seed = UInt64Array::from_iter(
+            Field::new("seed", DataType::UInt64),
+            std::iter::repeat(u64::MAX).map(Some).take(1),
+        )
+        .into_series();
+        macro_rules! gen_seed {
+            ($size:expr) => {
+                Some(seed.broadcast($size)?.u64()?)
+            };
+        }
+
+        let hashed_contains = contains
+            .hash_with_validity(gen_seed!(contains.len()))?
+            .get(0);
+
+        let bool_iter = self
+            .iter()
+            .map(|sub_series| -> DaftResult<_> {
+                let (sub_series, hashed_contains) = match sub_series.zip(hashed_contains) {
+                    Some(inner) => inner,
+                    None => return Ok(false),
+                };
+
+                let left = sub_series.hash_with_validity(gen_seed!(sub_series.len()))?;
+                let right = PrimitiveArray::from_slice([hashed_contains]);
+                let right_value = hashed_contains;
+                let comparator = build_is_equal(left.as_arrow(), &right, true, false)?;
+
+                Ok(left
+                    .into_iter()
+                    .flatten()
+                    .enumerate()
+                    .any(|(index, &left_value)| left_value == right_value && comparator(index, 0)))
+            })
+            .map(Some)
+            .map(Option::transpose);
```

Comment:
Since we are looping through every value in the series to do the hashing already, would it be faster just to directly compare every value in a list with `contains`?

Reply:
I believe in order to run comparisons, we would need to use the `with_match_comparable_daft_types!` macro. That one only operates on a limited number of data-types.

However, `.hash(..)` is implemented for more data-types than comparable supports. For example, we can hash structs. So if we hash the values first and turn them into `u64`s, we can perform comparison on many more types than simple comparison checking.

At least that's my reasoning. Seems like this is a tradeoff in terms of performance and API-availability.

Reply:
I think it's fine for lost.contains to only work for types in `with_match_comparable_daft_types!`. It's a constraint that other comparing expressions also have. Does `build_is_equal` use a different comparison method? Curious if it can handle the other data types and if so, why it can but other functions can't

Reply:
The `build_is_equal` may or may not be as constrained as other comparison functions. However, since I hash first and run `build_is_equal` on the hashed outputs, all I need is for `build_is_equal` to work for `u64`s.

Reply:
But I guess I agree with your take that `list.contains` should only work for things types which `with_match_comparable_daft_types!` currently supports. If it follows the pattern of what we do elsewhere, then that is the better behaviour for the end-users.

Reply:
I don't think you can rely on the hash function to be unique for every unique value though

Reply:
That's where the `let comparator = build_is_equal(..)` on L659 comes into play. If `left_value == right_value` (which may be true even if the underlying values are true) is true, then `comparator(index, 0)` is executed, which actually performs value comparison.

Actually, I'll look into how `comparator` performs value comparison in the first place. Maybe I can directly leverage that.

Reply:
Moved away from hash based implementation.